app/services/vendor_validator_service.rb

```diff
@@ -0,0 +1,45 @@
+class VendorValidatorService
```

Comment:
Can we find a better name for this, or split it up into multiple classes if necessary? Adding "Service" to the name of a class is a smell. It's implied that it's a service, so the name of the class should still make sense if you drop "Service" from its name, but `VendorValidator` would not be an appropriate name for what this class is doing.